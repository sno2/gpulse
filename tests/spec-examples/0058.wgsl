fn nested() -> i32 {
  var a: i32;
  {             // The start of a compound statement.
    a = 2;      // Behavior: {Next}
    return 1;   // Behavior: {Return}
  }             // The compound statement as a whole has behavior {Return}
  a = 1;        // Valid, statically unreachable code.
                //   Statement behavior: {Next}
                //   Overall behavior (due to sequential statements): {Return}
  return 2;     // Valid, statically unreachable code. Behavior: {Return}
}
