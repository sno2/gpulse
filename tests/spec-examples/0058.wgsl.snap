ast.Node{ .fn_decl = ast.FnDecl{ .name = ast.Span{ .start = 3, .end = 9 }, .params = {  }, .ret = ast.Node{ .identifier = ast.Span{ .start = 15, .end = 18 } }, .scope = { ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 27, .end = 28 }, .access_mode = null, .addr_space = null, .typ = ast.Node{ .identifier = ast.Span{ .start = 30, .end = 33 } }, .value = null } }, ast.Node{ .scope = ast.Scope{ .scope = { ast.Node{ .assign = ast.BinaryOp{ .lhs = ast.Node{ .identifier = ast.Span{ .start = 93, .end = 94 } }, .op_idx = 95, .rhs = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 97, .end = 98 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }, ast.Node{ .ret = ast.Ret{ .value = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 136, .end = 137 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } } } }, ast.Node{ .assign = ast.BinaryOp{ .lhs = ast.Node{ .identifier = ast.Span{ .start = 240, .end = 241 } }, .op_idx = 242, .rhs = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 244, .end = 245 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }, ast.Node{ .ret = ast.Ret{ .value = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 429, .end = 430 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } } } }
